cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureVirtualMachineService.java

```diff
@@ -51,6 +51,16 @@
         while (hasMissingVm(virtualMachines, privateInstanceIds) && virtualMachines.hasNextPage()) {
             virtualMachines.loadNextPage();
         }
+        if (hasNoVmInResourceGroup(virtualMachines)) {
```

Comment:
It might be worth logging that getting vms by resource group was unsuccessful, falling back to single vm queries.

Reply:
Good idea, I added more log message